rust/lance-io/src/object_store.rs

```diff
@@ -781,11 +807,15 @@ async fn configure_store(url: &str, options: ObjectStoreParams) -> Result<Object
             scheme: String::from("memory"),
             block_size: 64 * 1024,
         }),
-        unknow_scheme => {
-            let err = lance_core::Error::from(object_store::Error::NotSupported {
-                source: format!("Unsupported URI scheme: {}", unknow_scheme).into(),
-            });
-            Err(err)
+        unknown_scheme => {
+            if let Some(provider) = registry.providers.get(unknown_scheme) {
+                provider.new_store(url, &options)
+            } else {
+                let err = lance_core::Error::from(object_store::Error::NotSupported {
+                    source: format!("Unsupported URI scheme: {}", unknown_scheme).into(),
```

Comment:
I think it might be helpful in some contexts to provide the full URL in the error message so it's clearer where the scheme came from:
```suggestion
                    source: format!("Unsupported URI scheme: '{}' in URL '{}'", unknown_scheme, url).into(),
```